mitxgraders/helpers/calc.py

```diff
@@ -531,9 +531,19 @@ def eval_number(self, parse_result):
         """
         result = float(parse_result[0])
         if len(parse_result) == 2:
-            result = result * self.suffixes[parse_result[1]]
+            result *=  self.suffixes[parse_result[1]]
         return result
 
+    def eval_variable(self, parse_result):
+        """
+        Returns a copy of the variable in self.vars
+
+        NOTE: The variable's class must implement a __copy__ method.
+            (numpy ndarrays do implement this method)
```

Comment:
Add a note for future benefit - this is to make sure that the variables in the dictionary don't mutate!

Also, does the copy method applied to a MathArray produce another MathArray?

Finally, does IdentityMatrix have a copy method?

Reply:
I see that IdentityMatrix does indeed have copy.

Reply:
IdentityMultiple! =)